src/runtime/x86_cpu_features.cpp

```diff
@@ -79,6 +81,11 @@ WEAK CpuFeatures halide_get_cpu_features() {
             }
             if ((info2[1] & avx512_cannonlake) == avx512_cannonlake) {
                 features.set_available(halide_target_feature_avx512_cannonlake);
+                if ((info2[2] & avx512vnni) == avx512vnni) {
+                    // FIXME: Should also check AVX512-BF16 here, but that needs call to cpuid(eax=7, ecx=1)
+                    // FIXME: Should this check the LLVM version? Only supported on 12+.
+                    features.set_available(halide_target_feature_avx512_sapphirerapids);
```

Comment:
Hrm, that's a good question. I don't think that `LLVM_VERSION` is (currently) visible to our runtime code. That probably wouldn't be hard to fix (just minor changes in a couple of build scripts).

Reply:
I don't think this is necessary. It should just be a no-op if LLVM doesn't support it.

Reply:
To check my understanding: The code here will be run to check what CPU features are available when running already (AOT) compiled code. This cpu_features function lets the runtime choose which of the available AOT compiled functions to use, so if there is a sapphirerapids version available then that gets used, but if not then the next best target will be used. As this never leads to attempting to compile code it shouldn't matter what the version of LLVM is, only what AOT compiled functions are available.

Does that sound right?

Reply:
Yes.